modules/openxr/openxr_api.cpp

```diff
@@ -2577,6 +2618,14 @@ Size2 OpenXRAPI::get_play_space_bounds() const {
 	return ret;
 }
 
+PackedInt64Array OpenXRAPI::get_supported_swapchain_formats() {
+	PackedInt64Array supported_swapchain_list;
+	for (uint32_t i = 0; i < num_swapchain_formats; i++) {
+		supported_swapchain_list.push_back(supported_swapchain_formats[i]);
+	}
+	return supported_swapchain_list;
```

Comment:
Rather than building this array when requested, we could just store it as a `PackedInt64Array` (aka `Vector<int64_t>`) within `OpenXRAPI` - it would probably simplify the code in a few other places too.

It'd be something like [this diff](https://github.com/user-attachments/files/17549448/pr-97151-supported-swapchain-formats-vector.diff.txt) (untested).

Reply:
When most of that code was written the vector code in Godot wasn't as well thought out as it is today, so I'm all for making much more use of it instead of having our own memory management code in OpenXRAPI.

Reply:
Should be a separate PR though

Reply:
I am happy to update these in an additional PR! I've made a note of it and will resolve this conversation for now.